### DISCUSSION TITLE
Set Response Status Code in NextJS App Router

### DISCUSSION THREAD

**Discussion by GaChPaJ** [vercel/next.js#64500] [App Router]
How am I supposed to set the response status code in the following code?

`
import { notFound } from 'next/navigation'

async function BlogPostPage({ params }) {
    const blogPost = await fetchBlogPost({ slug: params.slug })

    if (!blogPost) {
        return notFound()
    }

    return <BlogPost blogPost={blogPost} />
}

export default BlogPostPage

`

The getServerSideProps approach does not work (explained here: https://github.com/vercel/next.js/issues/52914). Any ideas?

**Comment by SidKH** [comment]
The `notFound` automatically sets a 404 status code for non-streamed UI. 

When streaming (with loading.js or Suspense) the status code will always be 200. The Browsers API does not allow changing the status code after the streaming starts.

**Reply by votemike** [reply]
Is that mentioned in the docs anywhere?
So loading.js and Suspense aren't suitable to use if you want to use appropriate HTTP status codes? So probably no good for most standard websites.
What sort of applications would loading.js and Suspense be useful for?
